front/pages/w/[wId]/assistant/new.tsx

```diff
@@ -74,66 +82,80 @@ export const getServerSideProps = withDefaultUserAuthRequirements<
-  const globalAgentConfigurations = useAgentConfigurations({
-    workspaceId: owner.sId,
-    agentsGetView: "global",
-    includes: ["authors"],
-    sort: "priority",
-  }).agentConfigurations;
-
-  const workspaceAgentConfigurations = isBuilder
-    ? []
-    : useAgentConfigurations({
-        workspaceId: owner.sId,
-        agentsGetView: "workspace",
-        includes: ["authors"],
-        limit: 2,
-        sort: "alphabetical",
-      }).agentConfigurations;
-
-  const displayedAgents = [
-    ...globalAgentConfigurations,
-    ...workspaceAgentConfigurations,
-  ]
-    .filter((a) => a.status === "active")
-    // Sort is necessary due to separately fetched global and workspace assistants, ensuring unified ordering.
-    .sort(compareAgentsForSort)
-    .slice(0, isBuilder ? 2 : 4);
-
   const {
-    metadata: quickGuideSeen,
-    isMetadataError: isQuickGuideSeenError,
-    isMetadataLoading: isQuickGuideSeenLoading,
-    mutateMetadata: mutateQuickGuideSeen,
-  } = useUserMetadata("quick_guide_seen");
-  const [showQuickGuide, setShowQuickGuide] = useState<boolean>(false);
+    setAnimate,
+    setSelectedAssistant,
+    inputBarAssistants: agentConfigurations,
+    inputBarAssistantsLoading: isAgentConfigurationsLoading,
+  } = useContext(InputBarContext);
 
-  useEffect(() => {
-    if (!quickGuideSeen && !isQuickGuideSeenError && !isQuickGuideSeenLoading) {
-      // Quick guide has never been shown, lets show it.
-      setShowQuickGuide(true);
-    }
-  }, [isQuickGuideSeenError, isQuickGuideSeenLoading, quickGuideSeen]);
+  const agentsByTab = useMemo(() => {
+    const filteredAgents: LightAgentConfigurationType[] =
+      agentConfigurations.filter(
+        (a) =>
+          a.status === "active" &&
+          // Filters on search query
+          (assistantSearch.trim() === "" ||
+            subFilter(assistantSearch.toLowerCase(), a.name.toLowerCase()))
+      );
+
+    return {
+      all: filteredAgents,
+      published: filteredAgents.filter((a) => a.scope === "published"),
+      workspace: filteredAgents.filter((a) => a.scope === "workspace"),
+      personal: filteredAgents.filter((a) => a.scope === "private"),
+      // TODO: Implement most popular agents (upcoming PR for issue #5454)
+      most_popular: filteredAgents
+        .filter((a) => a.usage && a.usage.messageCount > 0)
+        .sort(
+          (a, b) => (b.usage?.messageCount || 0) - (a.usage?.messageCount || 0)
+        )
+        .slice(0, 0), // Placeholder -- most popular agents are not implemented yet
+    };
+  }, [agentConfigurations, assistantSearch]);
 
-  const { submit: handleSubmit } = useSubmitFunction(
+  const visibleTabs = useMemo(() => {
+    return ALL_AGENTS_TABS.filter((tab) => agentsByTab[tab.id].length > 0);
+  }, [agentsByTab]);
+
+  const [selectedTab, setSelectedTab] = useState<TabId | null>(
+    visibleTabs[0]?.id || null
+  );
+
+  const displayedTab = visibleTabs.find((tab) => tab.id === selectedTab)
```

Comment:
@flvndvd :wink: I used to agree with the no-ternary point of view, but here, it displays nicely I think and is quite readable (modulo the time to get used to the syntax of course, as with any language construct)